config/lib/seed.js

```diff
@@ -92,3 +65,85 @@ function reportError(err) {
   console.log(err);
   console.log();
 }
+
+function performSeedUser(user) {
+
+  var User = mongoose.model('User');
+
+  return new Promise(function(resolve, reject) {
+
+    var seedUser = {
+      username: 'user',
+      password: 'User_Password1!',
+      provider: 'local',
+      email: 'user@localhost.com',
+      firstName: 'User',
+      lastName: 'Local',
+      displayName: 'User Local',
+      roles: ['user']
+    };
+
+    var seedAdmin = {
+      username: 'admin',
+      password: 'Admin_Password1!',
+      provider: 'local',
+      email: 'admin@localhost.com',
+      firstName: 'Admin',
+      lastName: 'Local',
+      displayName: 'Admin Local',
+      roles: ['user', 'admin']
+    };
+
+    var user = null;
+
+    //If production only seed admin if it does not exist
+    if (process.env.NODE_ENV === 'production') {
+      //Add Local Admin
+
+      user = new User(seedAdmin);
+
+      User.generateRandomPassphrase()
```

Comment:
@mleanos the saveUser() must return a promise because it's doing an async operation.
If I leave the saveUser() as is in your original code then this breaks the whole point of the promises. The saveUser() function must be a thenable so I can chain it, that's why I refactored it to return a promise.